spec/parser/cfn_parser_ec2_instance_spec.rb

```diff
@@ -29,14 +25,23 @@
     it 'returns ec2 instance with security group' do
       yaml_test_templates('ec2_instance/instance_with_sgid_list_ref').each do |test_template|
         cfn_model = @cfn_parser.parse IO.read(test_template)
-
         ec2_instances = cfn_model.resources_by_type('AWS::EC2::Instance')
-
         expect(ec2_instances.size).to eq 1
         ec2_instance = ec2_instances.first
-
         expect(ec2_instance.security_groups.size).to eq 0
       end
     end
   end
-end
\ No newline at end of file
+
+  context 'an ec2 instance with a launch template', :ec2 do
+    it 'returns ec2 instance with launch template' do
+      yaml_test_templates('ec2_instance/instance_with_launch_template').each do |test_template|
+        cfn_model = @cfn_parser.parse IO.read(test_template)
+        ec2_launch_templates = cfn_model.resources_by_type('AWS::EC2::LaunchTemplate')
+        expect(ec2_launch_templates.size).to eq 1
+        ec2_instances = cfn_model.resources_by_type('AWS::EC2::Instance')
+        expect(ec2_instances.size).to eq 1
```

Comment:
the image id is nil now i suppose?  an explicit assertion here would capture that as the expected behavior